packaging/suse/portusctl/lib/configurator.rb

```diff
@@ -0,0 +1,186 @@
+      ]
+      Runner.exec("gensslcert", args)
+    end
+
+    FileUtils.chown("wwwrun", "www", "/etc/apache2/ssl.key")
+    FileUtils.chmod(750,"/etc/apache2/ssl.key")
+
+    FileUtils.chown("wwwrun", "www", "/etc/apache2/ssl.key/#{HOSTNAME}-server.key")
+    FileUtils.chmod(440,"/etc/apache2/ssl.key/#{HOSTNAME}-server.key")
+
+    # Create key used by Portus to sign the JWT tokens
+    FileUtils.ln_sf(
+      "/etc/apache2/ssl.key/#{HOSTNAME}-server.key",
+      File.join("/srv/Portus/config", "server.key"))
+
+    FileUtils.cp(
+      "/etc/apache2/ssl.crt/#{HOSTNAME}-ca.crt",
+      "/srv/www/htdocs")
+    FileUtils.chmod(755,"/srv/www/htdocs/#{HOSTNAME}-ca.crt")
+    FileUtils.cp(
+      "/etc/apache2/ssl.crt/#{HOSTNAME}-ca.crt",
+      "/etc/pki/trust/anchors")
+    Runner.exec("update-ca-certificates")
+  end
+
+  # Creates the database.yml file required by Rails
+  def database_config
+    template = File.join(
+      File.expand_path('../../templates', __FILE__),
+      'database.yml.erb')
+
+    conf = ERB.new(File.read(template), nil, '<>').result(binding)
+
+    File.open("/srv/Portus/config/database.yml", "w") do |file|
+      file.write(conf)
+    end
+  end
+
+  # Creates the database and performs the migrations
+  def create_database
+    env_variables = {
+      "SKIP_MIGRATION" => "yes",
+      "PORTUS_PASSWORD" => @portus_password
+    }
+    puts "Creating Portus' database"
+    Runner.bundler_exec("rake", ["db:create"], env_variables)
+
+    puts "Running database migrations"
+    Runner.bundler_exec("rake", ["db:migrate"], env_variables)
+
+    puts "Seeding database"
+    Runner.bundler_exec("rake", ["db:seed"], env_variables)
+  end
+
+  # Creates registry's configuration
+  def registry
+    # Add the certificated used by Portus to sign the JWT tokens
+    ssldir = "/etc/registry/ssl.crt"
+    FileUtils.mkdir_p(ssldir)
+    FileUtils.ln_sf(
+      "/etc/apache2/ssl.crt/#{HOSTNAME}-server.crt",
+      File.join(ssldir, "portus.crt"))
+
+    template = File.join(
+      File.expand_path('../../templates', __FILE__),
+      'registry.yml.erb')
+
+    conf = ERB.new(File.read(template), nil, '<>').result(binding)
+
+    File.open("/etc/registry/config.yml", "w") do |file|
+      file.write(conf)
+    end
+    Runner.exec("configure_registry.sh")
+  end
+
+  # Creates the config-local.yml file used by Portus
+  def config_local
+    if Process.uid != 0
+      warn "Apache configuration must run as root user"
+      exit 1
+    end
+
+    template = File.join(
+      File.expand_path('../../templates', __FILE__),
+      'config-local.yml.erb')
+    template = ERB.new(File.read(template), nil, '<>').result(binding)
+    File.open("/srv/Portus/config/config-local.yml", "w") do |file|
+      file.write(template)
+    end
+  end
+
+  # Creates the secrets.yml file used by Rails
+  def secrets
+    template = File.join(
+      File.expand_path('../../templates', __FILE__),
+      'secrets.yml.erb')
```

Comment:
TODO: secrets.yml should not be owned by the package because we are modifying/creating it